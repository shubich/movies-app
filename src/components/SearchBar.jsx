import React from 'react';
import { Link } from 'react-router-dom';

export class SearchBar extends React.Component {
    constructor(props) {
        super(props);
        if (this.props.initialVal) {
            this.props.getFilms(this.props.initialVal)
        }
    }

    handleSubmit = (e) => {
        e.preventDefault();
        if (this.searchQuery.value) {
            this.props.history.push(`/search/${this.searchQuery.value}`);
            this.props.getFilms(this.searchQuery.value)
        } else {
            this.searchQuery.focus();
        }
    }

    setRef = (node) => {
        if (node) this[node.name] = node;
    } 

    render() {
        return (
            <div>
                <div id='search-bar'>
                    <h2>FIND YOUR MOVIE</h2>
                    <form onSubmit={this.handleSubmit}>
                        <input 
                            type='text' 
                            name='searchQuery'
                            autoComplete='off'
                            defaultValue={this.props.initialVal}
                            placeholder='Pulp Fiction'
                            ref={this.setRef}
                        />
                        <div id='search-props'>
                            <span>SEARCH BY</span>
                            <button className='btn btn-primary'>TITLE</button>
                            <button className='btn btn-basic'>DIRECTOR</button>
                        </div>
                        <button type='submit' className='btn btn-primary btn-search'>SEARCH</button>
                    </form>
                </div>
            </div>
        )
    }
}