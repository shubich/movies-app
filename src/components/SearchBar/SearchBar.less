@import "../../styles/palette";
@import "../../styles/media";

.SearchBar {
    display: flex;
    flex-flow: column wrap;
}

.SearchBar-Form {
    display: flex;
    flex-flow: row wrap;
    input[type='text'] {
        flex-basis: 100%;
        padding: 10px;
        box-sizing: border-box;
        border: 5px solid #111111;
        border-bottom-color: @primary-color;
        box-shadow: 1px 1px 5px #333;
        margin-bottom: 15px;
        &:focus {
            border-color: @primary-color;
        }
    }
}

.SearchBar-Props {
    flex: 1;
    margin-right: 15px;
}

.SearchBar-RadioBtn {
    margin-left: 10px;
    cursor: default;
    font-size: 13.333px;
    display: inline-block;
    padding: 3px 9px;
    border: none;
    box-shadow: 1px 1px 5px #444;
    input {
        display: none;
    }
}

.SearchBar-RadioBtn_primary {
    background: @primary-color;
    color: @default-color;
}

.SearchBar-RadioBtn_basic {
    background: gray;
    color: @default-color;
}

@media (max-width: 525px) {
    .SearchBar-SearchBy {
        display: block;
    }
    
    .SearchBar-RadioBtn {
        margin-left: 0;
    }
}
