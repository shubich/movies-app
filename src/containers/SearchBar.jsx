import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import getFilms from '../actions/getFilms';

class SearchBar extends React.Component {
    constructor(props) {
        super(props);
        this.initialTitle = this.props.match.url.split('/')[2];
        if (this.initialTitle) {
            this.props.dispatch(getFilms(this.initialTitle))
        }
    }

    handleSubmit = (e) => {
        e.preventDefault();
        if (this.searchQuery.value) {
            this.props.history.push(`/search/${this.searchQuery.value}`);
            this.props.dispatch(getFilms(this.searchQuery.value))
        } else {
            this.searchQuery.focus();
        }
    }

    render() {
        return (
            <div>
                <div id='search-bar'>
                    <h2>FIND YOUR MOVIE</h2>
                    <form onSubmit={this.handleSubmit}>
                        <input 
                            type='text' 
                            defaultValue={this.initialTitle}
                            placeholder='Pulp Fiction'
                            ref={node => {this.searchQuery = node}}
                        />
                        <div id='search-props'>
                            <span>SEARCH BY</span>
                            <button className='btn btn-primary'>TITLE</button>
                            <button className='btn btn-basic'>DIRECTOR</button>
                        </div>
                        <button type='submit' className='btn btn-primary btn-search'>SEARCH</button>
                    </form>
                </div>
            </div>
        )
    }
} 

export default SearchBar = connect()(SearchBar);