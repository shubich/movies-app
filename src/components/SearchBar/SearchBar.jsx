import React from 'react';
import PropTypes from 'prop-types';
import Button from '../Button';

import './SearchBar.less';

export default class SearchBar extends React.Component {
  componentWillReceiveProps(nextProps) {
    this.searchLine.value = nextProps.searchQuery || '';
  }

  setRef = (node) => {
    this.searchLine = node;
  }

  handleSubmit = (e) => {
    e.preventDefault();
    if (this.searchLine.value) {
      const searchBy = this.props.searchType;
      this.props.history.push(`/search/${searchBy}/${this.searchLine.value}`);
    } else {
      this.searchLine.focus();
    }
  }

  handleOptionChange = (changeEvent) => {
    const searchBy = changeEvent.target.value;
    this.props.history.push(`/search/${searchBy}/${this.searchLine.value}`);
  }

  render() {
    return (
      <div className="SearchBar">
        <h2>FIND YOUR MOVIE</h2>
        <form className="SearchBar-Form" onSubmit={this.handleSubmit}>
          <input
            type="text"
            name="searchLine"
            autoComplete="off"
            defaultValue={this.props.searchQuery}
            placeholder="Pulp Fiction"
            ref={this.setRef}
          />
          <div className="SearchBar-Props">
            <span>SEARCH BY</span>

            <label className={`SearchBar-RadioBtn SearchBar-RadioBtn_${this.props.searchType === 'movies' ? 'primary' : 'basic'}`}>
              <input
                type="radio"
                value="movies"
                checked={this.props.searchType === 'movies'}
                onChange={this.handleOptionChange}
              />
                TITLE
            </label>

            <label className={`SearchBar-RadioBtn SearchBar-RadioBtn_${this.props.searchType === 'people' ? 'primary' : 'basic'}`}>
              <input
                type="radio"
                value="people"
                checked={this.props.searchType === 'people'}
                onChange={this.handleOptionChange}
              />
                DIRECTOR
            </label>
          </div>
          <Button type="submit" primary search>SEARCH</Button>
        </form>
      </div>
    );
  }
}

SearchBar.propTypes = {
  searchQuery: PropTypes.string,
  searchType: PropTypes.string.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func,
  }).isRequired,
};

SearchBar.defaultProps = {
  searchQuery: '',
};

