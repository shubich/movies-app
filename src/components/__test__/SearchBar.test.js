import React from 'react';
import { SearchBar } from '../SearchBar';
import { shallow } from 'enzyme';
import { mount } from 'enzyme';

let props;

describe('SearchBar', () => {
    beforeEach(() => {
        props = {
            initialVal: 2330,
            getFilms: jest.fn(),
            getPeople: jest.fn(),
            setSearchType: jest.fn(),
            history: { push: jest.fn() },
            searchType: 'title' 
        };
    });

    it('renders correctly with props', () => {
        const wrapper = shallow(
            <SearchBar {...props} />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with another search type', () => {
        const wrapper = shallow(
            <SearchBar {...props} searchType='person' />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should call getFilms', () => {
        const wrapper = mount(
            <SearchBar {...props} />
        );
        // const wrapperInstance = wrapper.instance(); // --shallow
        // wrapperInstance.searchQuery={value: 'something'}; // --shallow
        wrapper.find('form').simulate('submit', {preventDefault:jest.fn()})
        expect(props.getFilms).toBeCalledWith('2330');
    });

    it('should call getPeople', () => {
        const wrapper = mount(
            <SearchBar {...props} searchType={'person'} />
        );
        wrapper.find('form').simulate('submit', {preventDefault:jest.fn()})
        expect(props.getPeople).toBeCalledWith('2330');
    });

    it('should call input.focus()', () => {
        const focus = jest.fn();
        const wrapper = shallow(
            <SearchBar {...props} initialVal={''} />
        );
        const wrapperInstance = wrapper.instance(); // --shallow
        wrapperInstance.searchQuery={ focus }; // --shallow
        
        wrapper.find('form').simulate('submit', {preventDefault:jest.fn()})
        expect(focus).toBeCalled();
    });

    it('should call setSearchType on radio changes', () => {
        const wrapper = mount(
            <SearchBar {...props} />
        );
        
        wrapper.find('[checked=false]').simulate('change', {target: {value: 'test'}});
        expect(props.setSearchType).toBeCalledWith('test');
    })
})