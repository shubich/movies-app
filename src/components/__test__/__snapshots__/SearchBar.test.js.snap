// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders correctly with props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar
    getFilms={[Function]}
    history={[Function]}
    initialVal={2330}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <div
        id="search-bar"
      >
        <h2>
          FIND YOUR MOVIE
        </h2>
        <form
          onSubmit={[Function]}
        >
          <input
            autoComplete="off"
            defaultValue={2330}
            name="searchQuery"
            placeholder="Pulp Fiction"
            type="text"
          />
          <div
            id="search-props"
          >
            <span>
              SEARCH BY
            </span>
            <button
              className="btn btn-primary"
            >
              TITLE
            </button>
            <button
              className="btn btn-basic"
            >
              DIRECTOR
            </button>
          </div>
          <button
            className="btn btn-primary btn-search"
            type="submit"
          >
            SEARCH
          </button>
        </form>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            FIND YOUR MOVIE
          </h2>,
          <form
            onSubmit={[Function]}
          >
            <input
              autoComplete="off"
              defaultValue={2330}
              name="searchQuery"
              placeholder="Pulp Fiction"
              type="text"
            />
            <div
              id="search-props"
            >
              <span>
                SEARCH BY
              </span>
              <button
                className="btn btn-primary"
              >
                TITLE
              </button>
              <button
                className="btn btn-basic"
              >
                DIRECTOR
              </button>
            </div>
            <button
              className="btn btn-primary btn-search"
              type="submit"
            >
              SEARCH
            </button>
          </form>,
        ],
        "id": "search-bar",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "FIND YOUR MOVIE",
          },
          "ref": null,
          "rendered": "FIND YOUR MOVIE",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                autoComplete="off"
                defaultValue={2330}
                name="searchQuery"
                placeholder="Pulp Fiction"
                type="text"
              />,
              <div
                id="search-props"
              >
                <span>
                  SEARCH BY
                </span>
                <button
                  className="btn btn-primary"
                >
                  TITLE
                </button>
                <button
                  className="btn btn-basic"
                >
                  DIRECTOR
                </button>
              </div>,
              <button
                className="btn btn-primary btn-search"
                type="submit"
              >
                SEARCH
              </button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "autoComplete": "off",
                "defaultValue": 2330,
                "name": "searchQuery",
                "placeholder": "Pulp Fiction",
                "type": "text",
              },
              "ref": [Function],
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <span>
                    SEARCH BY
                  </span>,
                  <button
                    className="btn btn-primary"
                  >
                    TITLE
                  </button>,
                  <button
                    className="btn btn-basic"
                  >
                    DIRECTOR
                  </button>,
                ],
                "id": "search-props",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "SEARCH BY",
                  },
                  "ref": null,
                  "rendered": "SEARCH BY",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "TITLE",
                    "className": "btn btn-primary",
                  },
                  "ref": null,
                  "rendered": "TITLE",
                  "type": "button",
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "host",
                  "props": Object {
                    "children": "DIRECTOR",
                    "className": "btn btn-basic",
                  },
                  "ref": null,
                  "rendered": "DIRECTOR",
                  "type": "button",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "SEARCH",
                "className": "btn btn-primary btn-search",
                "type": "submit",
              },
              "ref": null,
              "rendered": "SEARCH",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <div
          id="search-bar"
        >
          <h2>
            FIND YOUR MOVIE
          </h2>
          <form
            onSubmit={[Function]}
          >
            <input
              autoComplete="off"
              defaultValue={2330}
              name="searchQuery"
              placeholder="Pulp Fiction"
              type="text"
            />
            <div
              id="search-props"
            >
              <span>
                SEARCH BY
              </span>
              <button
                className="btn btn-primary"
              >
                TITLE
              </button>
              <button
                className="btn btn-basic"
              >
                DIRECTOR
              </button>
            </div>
            <button
              className="btn btn-primary btn-search"
              type="submit"
            >
              SEARCH
            </button>
          </form>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2>
              FIND YOUR MOVIE
            </h2>,
            <form
              onSubmit={[Function]}
            >
              <input
                autoComplete="off"
                defaultValue={2330}
                name="searchQuery"
                placeholder="Pulp Fiction"
                type="text"
              />
              <div
                id="search-props"
              >
                <span>
                  SEARCH BY
                </span>
                <button
                  className="btn btn-primary"
                >
                  TITLE
                </button>
                <button
                  className="btn btn-basic"
                >
                  DIRECTOR
                </button>
              </div>
              <button
                className="btn btn-primary btn-search"
                type="submit"
              >
                SEARCH
              </button>
            </form>,
          ],
          "id": "search-bar",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "FIND YOUR MOVIE",
            },
            "ref": null,
            "rendered": "FIND YOUR MOVIE",
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <input
                  autoComplete="off"
                  defaultValue={2330}
                  name="searchQuery"
                  placeholder="Pulp Fiction"
                  type="text"
                />,
                <div
                  id="search-props"
                >
                  <span>
                    SEARCH BY
                  </span>
                  <button
                    className="btn btn-primary"
                  >
                    TITLE
                  </button>
                  <button
                    className="btn btn-basic"
                  >
                    DIRECTOR
                  </button>
                </div>,
                <button
                  className="btn btn-primary btn-search"
                  type="submit"
                >
                  SEARCH
                </button>,
              ],
              "onSubmit": [Function],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "autoComplete": "off",
                  "defaultValue": 2330,
                  "name": "searchQuery",
                  "placeholder": "Pulp Fiction",
                  "type": "text",
                },
                "ref": [Function],
                "rendered": null,
                "type": "input",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <span>
                      SEARCH BY
                    </span>,
                    <button
                      className="btn btn-primary"
                    >
                      TITLE
                    </button>,
                    <button
                      className="btn btn-basic"
                    >
                      DIRECTOR
                    </button>,
                  ],
                  "id": "search-props",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "SEARCH BY",
                    },
                    "ref": null,
                    "rendered": "SEARCH BY",
                    "type": "span",
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "TITLE",
                      "className": "btn btn-primary",
                    },
                    "ref": null,
                    "rendered": "TITLE",
                    "type": "button",
                  },
                  Object {
                    "instance": null,
                    "key": null,
                    "nodeType": "host",
                    "props": Object {
                      "children": "DIRECTOR",
                      "className": "btn btn-basic",
                    },
                    "ref": null,
                    "rendered": "DIRECTOR",
                    "type": "button",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "SEARCH",
                  "className": "btn btn-primary btn-search",
                  "type": "submit",
                },
                "ref": null,
                "rendered": "SEARCH",
                "type": "button",
              },
            ],
            "type": "form",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
